### PR TITLE
Uint: Add support for bit and,or,xor assign traits

### LINKED ISSUE #401: Help with primitive-types
I am struggling with primitive-types:

```
cannot use `|=` on type `primitive_types::U256`
```

Is there a work around for using these combined operators? Also is there a simple way to flip a bit at a given position? Thanks!

### PR DESCRIPTION
**Pull request by halo3mic**
Add support uint support for traits `BitAndAssign`, `BitXorAssign ` and `BitOrAssign `.

Fixes #401

### CODE REVIEW

**Review by ordian — APPROVED** [review]
Thanks! Could you add some tests for the added methods, please?

**Comment by halo3mic** [comment]
> Thanks! Could you add some tests for the added methods, please?

Sure, are the ones I just added okay?

File: uint/tests/uint_tests.rs [hunk 1]
```diff
@@ -1151,6 +1151,47 @@ fn trailing_zeros() {
...
+			x |= b;
+			assert_eq!(x, a | b);
+		}
+		// xor
+		{
+			let mut x = a;
+			x ^= b;
+			assert_eq!(x, a ^ b);
+		}
+		// shr
```
Comment (niklasad1) [inline uint/tests/uint_tests.rs:1175]: It looks like `shr` and `shl` were already implemented and you just added tests for them?

**Review by niklasad1 — APPROVED** [review]

File: uint/tests/uint_tests.rs (hunk 1 above)
Comment (halo3mic) [reply uint/tests/uint_tests.rs:1175]: That is correct, I couldn't find the tests for them and figured it wouldn't hurt to add them.

**Comment by AaronKutch** [comment]
Just FYI you broke semver and one of my crates with this

**Comment by ordian** [comment]
@AaronKutch could you share what broke in 0.9.5 compared to 0.9.4? This is the only change there.

Or did you mean that some version in the 0.9.x series was semver breaking? In that case I would suspect 0.9.2 https://github.com/paritytech/parity-common/blob/master/uint/CHANGELOG.md#092---2022-01-28 breaking MSRV, which we I personally consider non-breaking, but some people disagree.

**Comment by AaronKutch** [comment]
I had implemented some traits for the struct from `uint::construct_uint!` that the macro did not implement before, and after some 0.9 patch there was a trait collision. I guess I am breaking MSRV sometimes myself, but adding traits is more likely to cause problems in the circumstance of structs from macros that can circumvent the orphan rule.

**Comment by ordian** [comment]
I see. Thanks for the clarification.

If we follow that logic, then adding an inherent method would also be considered breaking, because a user could have added a method with the same name as well.

The docs says adding adding an inherent item is considered a "possibly-breaking" change (not for the macro case though): https://doc.rust-lang.org/cargo/reference/semver.html#possibly-breaking-change-adding-any-inherent-items.

I'll consider major release for such changes next time. Thanks for bringing this to my attention :)

**Pull request merged**